concordium-node/src/connection/low_level.rs

```diff
@@ -441,7 +441,9 @@ impl ConnectionLowLevel {
                     }
                 }
 
-                self.socket_buffer.reset();
+                // If we reach this point we have read the entire noise message.
+                // So consume it from the socket buffer.
```

Comment:
You are here describing what is going on in the code below. A better comment would be describing **why**, i.e. there may still messages left in the buffer for us to read.

Also, isn't this only ever going to be a thing for the responder? As the cause of this, is that the initiator sends a message before the responder recognized he is past the handshake. 

Also, this will explain why the `shift` is not happening in the `if i'm responder` clause.

Reply:
Ok, I'll try to add a bit more.

The thing is, that it is kind of "obvious" that the new behaviour is the correct one. We read a message, we consume it from the buffer.

So in that sense there is not really a need to explain it in terms of the bug that we had, since that was just wrong behaviour. But I'll add more context.

Reply:
I have added a bit of documentation above next to where we consume from the socket buffer for post-handshake connections only.